sdk/src/main/java/com/karhoo/sdk/analytics/Payloader.kt

```diff
@@ -79,9 +87,75 @@ class Payloader internal constructor(builder: Builder) {
                 payload[DESTINATION_ADDRESS] = it.displayAddress
             }
             payload[IS_PREBOOK] = tripDetails.dateScheduled != null
-            if (outboundTripId != null && !outboundTripId.isEmpty()) {
+            if (outboundTripId != null && outboundTripId.isNotEmpty()) {
                 payload[OUTBOUND_TRIP_ID] = outboundTripId
             }
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+
+            return this
+        }
+
+        fun bookingSuccess(
+            tripDetails: TripInfo,
+            correlationId: String?,
+            quoteId: String?
+        ): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+            payload[TRIP_ID] = tripDetails.tripId
+
+            return this
+        }
+
+        fun bookingFailure(
+            correlationId: String?,
+            quoteId: String?,
+            errorMessage: String,
+            lastFourDigits: String,
+            date: Date,
+            amount: Int,
+            currency: String
+        ): Builder {
+            quoteId?.let {
+                payload[QUOTE_ID] = it
+            }
+            correlationId?.let {
+                payload[CORRELATION_ID] = it
+            }
+            payload[PAYMENT_ERROR_MESSAGE] = errorMessage
+            payload[PAYMENT_CARD_LAST_FOUR_DIGITS] = lastFourDigits
+            payload[PAYMENT_AMOUNT] = amount
+            payload[PAYMENT_CURRENCY] = currency
+            payload[PAYMENT_DATE] = date.toString()
+
+            return this
```

Comment:
@BartlomiejSopalaKarhoo The spreadsheet contains an extra argument for payment method used which I don't see here. Did you and Matei agree on leaving it out on both platforms or is this an oversight?